src/htmx.js

```diff
@@ -2955,9 +2943,29 @@ var htmx = (function() {
       cleanUpElement(elt)
       return
     }
-    initNode(elt)
-    forEach(findElementsToProcess(elt), function(child) { initNode(child) })
+
+    const elementsToInit = []
+
+    function maybeDeInit(e) {
+      if (closest(e, htmx.config.disableSelector)) {
+        cleanUpElement(e)
+        return
+      }
+      const nodeData = getInternalData(e)
+      const hash = attributeHash(e)
+      if (nodeData.initHash !== hash) {
+        deInitNode(e)
+        nodeData.initHash = hash
+        elementsToInit.push(e)
+      }
+    }
+
+    maybeDeInit(elt)
+    forEach(findElementsToProcess(elt), maybeDeInit)
+
     forEach(findHxOnWildcardElements(elt), processHxOnWildcard)
+    initNode(elt)
```

Comment:
This is being done unconditionally, where previously it was unconditional.
Plus `elt` would already be in `elementsToInit` if needed, so this line should be removed entirely.

Reply:
Good catch, fixed!